lib/honeybadger/notice.rb

```diff
@@ -212,10 +212,18 @@ def ignore_by_class?(ignored_class = nil)
       ignored_class ? @ignore_by_class.call(ignored_class) : @ignore_by_class
     end
 
+    # Ignore exceptions based on Sidekiq job retry count. 
+    def ignore_by_sidekiq_attempts 
+      threshold = ::Honeybadger.configuration.sidekiq_job_attempt_threshold.to_i || 0
+      retry_count = parameters['retry_count'].to_i || 0
```

Comment:
@cdale77 sorry for the delay in getting back to you. I just released a major upgrade to our gem, so this will need to be ported to 2.0 and a new PR submitted. I think your implementation is good, and the new plugin system may have some advantage (the configuration is available inside the "execution" block rather than having to call Honeybadger.configuration globally, for instance.)

I'd like to work this into the next minor release, which will be 2.1. I'd accept a new PR if you want to tackle it, but I'm happy to port your changes over myself and credit you in the changelog. Either way, thank you for your contributions. :)